src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +150,52 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            byte[]? bytes = null;
```

Comment:
```suggestion
            byte[] bytes =  ArrayPool<byte>.Shared.Rent(4096);
```

Reply:
This is required in the `finally` block, that's why I had to hoist it out of `try`.

Reply:
There is no harm in initializing it before the try.

Reply:
(It can stay as is if you prefer. This is minor nit.)

Reply:
True, moved it outside of try and removed the null check in finally.